designer/server/src/main/scala/pl/touk/nussknacker/ui/api/BaseHttpService.scala

```diff
@@ -0,0 +1,60 @@
+package pl.touk.nussknacker.ui.api
+
+import com.typesafe.config.Config
+import pl.touk.nussknacker.ui.process.ProcessCategoryService
+import pl.touk.nussknacker.ui.security.api._
+import sttp.tapir.AnyEndpoint
+import sttp.tapir.server.ServerEndpoint
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.{ExecutionContext, Future}
+
+abstract class BaseHttpService(config: Config,
+                               processCategoryService: ProcessCategoryService,
+                               authenticator: AuthenticationResources)
+                              (implicit executionContext: ExecutionContext) {
+
+  private val allServerEndpoints = new AtomicReference(List.empty[ServerEndpoint[Any, Future]])
+
+  protected def expose(serverEndpoint: ServerEndpoint[Any, Future]): Unit = {
```

Comment:
at this stage, there is no sense to do it in a different way. `SeverEndpoint[Any, Future]` is an equivalent of Akka's `Route`. We don't need types here any more (see [docs](https://tapir.softwaremill.com/en/latest/server/logic.html?highlight=%20ServerEndpoint%5BAny%2C))

Reply:
But in the case of `Route` you still have types but secured by type classes - you have to provide `Marshaller` for your return type. I'll dig more in the documentation

Reply:
Marshaller is equivalent to Tapir's `Codec`. In the case of Akka HTTP it was created from Circe's encoder. In Tapir we have a mentioned `Codec` and Circe's codec too.

So, I don't understand what do you want to achieve here. IMO there is no problem with the `Any` in this place. I would do it like this:
```
protected def expose[T](serverEndpoint: ServerEndpoint[T, Future]): Unit
```
But it won't change anything.

Could you please explain exactly what you would like to do?

Reply:
When you take a look at `AkkaHttpServerInterpreter` class you will see the `def toRoute(ses: List[ServerEndpoint[AkkaStreams with WebSockets, Future]])` method. The first generic param is the capabilities type. We put in the invocation of the method the `serverEndpoint`s collected by our `expose` method. Because we don't use any special capabilities, we say "meh, I don't use ANY special capability". Akka server says: "I can handle `AkkaStreams with WebSockets` capability, but if you don't use any special one, I can handle your route/endpoint". 

That's why all is good here :)

Reply:
My "only" problem is that without knowledge of tapir I saw some mysterious `Any` and I don't know if it can be String, Boolean or whatever there. Or maybe it can be both String and Boolean simultaneously? Now I understand that they have contravariant "Requirement" and Any means that you don't require anything. We can leave it as it is, maybe I have to learn more about tapir's concept. Another option is to add a type alias e.g. `NoRequirementServerEndpoint = ServerEndpoint[Any, Future]` and some little comment next to it